v1/remote/transport/transport.go

```diff
@@ -53,7 +54,11 @@ func New(ref name.Reference, auth authn.Authenticator, t http.RoundTripper, a Sc
 	case anonymous:
 		return t, nil
 	case basic:
-		return &basicTransport{inner: t, auth: auth}, nil
+		u, err := url.Parse(ref.Context().Registry.String())
+		if err != nil {
+			return nil, err
+		}
+		return &basicTransport{inner: t, auth: auth, target: u.Host}, nil
```

Comment:
Why isn't this just `target: ref.Context().RegistryStr()`?

Reply:
Mea culpa